### DISCUSSION TITLE
Blog post on packaging 2021

### DISCUSSION THREAD

**Discussion by henryiii** [pypa/cibuildwheel#565] [General]
Here's a useful blogpost on the state of packaging in 2021:

https://labs.quansight.org/blog/2021/01/python-packaging-brainstorm/

Sadly don't even mention cibuildwheel, but lots of useful things and mentions there.

**Comment by henryiii** [comment]
I found this very likely useful: SciPy provides a metapackage with oldest NumPy supported on various systems for building. https://github.com/scipy/oldest-supported-numpy/blob/master/setup.cfg 

Great for Cython projects, perhaps? @deepcharles This might simplify things? Though it's 3.5+ only, so anyone supporting 2.7 will have to list both numpy 2.7 and this metapackage for 3+ with environment markers still. I'll probably try it out on a couple of our Cython packages and update the page here in the near future: https://scikit-hep.org/developer/packaging#special-additions-numpy

**Reply by deepcharles** [reply]
Thanks, this is indeed very useful. We have added it in [this PR](https://github.com/deepcharles/ruptures/pull/114).

**Reply by henryiii** [reply]
We are updating our example with this too in #567  :)
